samples/quickstart.js

```diff
@@ -117,25 +108,31 @@ async function main() {
       .on('error', console.error)
       .on('data', data => {
         try {
-          const decodedData = avroType.decode(
-            data.avroRows.serializedBinaryRows
-          );
+          offset = 0;
+
+          // Decode all rows in buffer
+          do {
+            const decodedData = avroType.decode(
+              data.avroRows.serializedBinaryRows,
+              offset
+            );
 
-          names.add(decodedData.value.name);
+            if (decodedData.value) {
+              names.add(decodedData.value.name);
 
-          if (!states[decodedData.value.state]) {
-            states[decodedData.value.state] = true;
-          }
+              if (!states.includes(decodedData.value.state)) {
+                states.push(decodedData.value.state);
+              }
+            }
 
-          offset = decodedData.offset;
+            offset = decodedData.offset;
+          } while (offset > 0);
```

Comment:
This while condition seems odd to me, but I may simply be unfamiliar with the decoder idioms in the node avro libraries.  offset == bytes remaining or something similar?

Reply:
You are right, updated to use a separate variable for tracking position when decoding rows